tools/integration_tests/gzip/gzip_test.go

```diff
@@ -0,0 +1,173 @@
+		keepCacheControlNoTransform bool // if true, no-transform is reset as ''
+		enableGzipEncodedContent    bool // if true, original file content is gzip-encoded
+		enableGzipContentEncoding   bool // if true, the content is uploaded as gsutil cp -Z i.e. with content-encoding: gzip header in GCS
+	}{
+		{
+			filename:                    TextContentWithContentEncodingWithNoTransformFilename,
+			filesize:                    TextContentSize,
+			keepCacheControlNoTransform: true,
+			enableGzipEncodedContent:    false,
+			enableGzipContentEncoding:   true,
+		},
+		{
+			filename:                    TextContentWithContentEncodingWithoutNoTransformFilename,
+			filesize:                    TextContentSize,
+			keepCacheControlNoTransform: false,
+			enableGzipEncodedContent:    false,
+			enableGzipContentEncoding:   true,
+		},
+		{
+			filename:                    GzipContentWithoutContentEncodingFilename,
+			filesize:                    TextContentSize,
+			keepCacheControlNoTransform: true, // it's a don't care in this case
+			enableGzipEncodedContent:    true,
+			enableGzipContentEncoding:   false,
+		}, {
+			filename:                    GzipContentWithContentEncodingWithNoTransformFilename,
+			filesize:                    TextContentSize,
+			keepCacheControlNoTransform: true,
+			enableGzipEncodedContent:    true,
+			enableGzipContentEncoding:   true,
+		}, {
+			filename:                    GzipContentWithContentEncodingWithoutNoTransformFilename,
+			filesize:                    TextContentSize,
+			keepCacheControlNoTransform: false,
+			enableGzipEncodedContent:    true,
+			enableGzipContentEncoding:   true,
+		},
+	}
+
+	for _, fmd := range fmds {
+		var localFilePath string
+		localFilePath, err := helpers.CreateLocalTempFile(fmd.filesize, fmd.enableGzipEncodedContent)
+		if err != nil {
+			return err
+		}
+
+		defer os.Remove(localFilePath)
+
+		// upload to the test-bucket for testing
+		gcsObjectPath := path.Join(setup.TestBucket(), GzipTestResPath, fmd.filename)
+
+		err = helpers.UploadGcsObject(localFilePath, gcsObjectPath, fmd.enableGzipContentEncoding)
+		if err != nil {
+			return err
+		}
+
+		gcsObjectsToBeDeletedEventually = append(gcsObjectsToBeDeletedEventually, gcsObjectPath)
+
+		if !fmd.keepCacheControlNoTransform {
+			err = helpers.ClearCacheControlOnGcsObject(gcsObjectPath)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func destroy_testdata(m *testing.M) error {
+	for _, gcsObjectPath := range gcsObjectsToBeDeletedEventually {
+		err := helpers.DeleteGcsObject(gcsObjectPath)
+		if err != nil {
+			return fmt.Errorf("Failed to delete gcs object gs://%s", gcsObjectPath)
+		}
+	}
+
+	return nil
+}
+
+func TestMain(m *testing.M) {
+	setup.ParseSetUpFlags()
+
+	commonFlags := "--sequential-read-size-mb=" + fmt.Sprint(SeqReadSizeMb) //+ " --implicit-dirs"
+	flags := [][]string{{commonFlags}}
+	// flags := [][]string{{commonFlags}, {commonFlags, "--enable-storage-client-library=false"}}
```

Comment:
`--enable-storage-client-library=false` fails with the same error as in  https://github.com/GoogleCloudPlatform/gcsfuse/pull/1256/commits/3ac6df6dc005e94f3a2c34964d7d76747bb60f03#r1282722070

Reply:
This is now fixed.